etl-pipeline/processes/record_clusterer.py

```diff
@@ -228,11 +309,27 @@ def _titles_overlap(
         return True
 
     def _tokenize_title(self, title: str):
+        """Converts a title string into a set of normalized "tokens" (words).
+        
+        1. Extracts words using regex
+        2. Converts to lowercase
+        3. Removes common stop words
+        
+        Returns:
+            set[str]: Set of normalized title words
+        """
         title_tokens = re.findall(r"(\w+)", title.lower())
 
         return set(title_tokens) - set(["a", "an", "the", "of"])
 
     def _format_identifiers(self, identifiers: list[str]):
+        """Formats identifiers for PostgreSQL array overlap query.
+        
+        Handles escaping of special characters and creates proper array syntax.
+        
+        Returns:
+            str: PostgreSQL-compatible array string
```

Comment:
same here - let's add -> str for the type hint. we can keep it since it adds some details on the significance of that string.